sdk/storage/azure-storage-common/inc/azure/storage/common/internal/xml_wrapper.hpp

```diff
@@ -37,44 +37,39 @@ namespace Azure { namespace Storage { namespace _internal {
 
   class XmlReader final {
   public:
-    explicit XmlReader(const char* data, size_t length);
+    XmlReader(const char* data, size_t length);
     XmlReader(const XmlReader& other) = delete;
     XmlReader& operator=(const XmlReader& other) = delete;
-    XmlReader(XmlReader&& other) noexcept { *this = std::move(other); }
-    XmlReader& operator=(XmlReader&& other) noexcept
-    {
-      m_context = other.m_context;
-      other.m_context = nullptr;
-      return *this;
-    }
+    XmlReader(XmlReader&& other) noexcept;
+    XmlReader& operator=(XmlReader&& other) noexcept;
```

Comment:
Is there a reason to move the move and copy operators out of line from the header?

Reply:
Actually yes. To be precise, the move constructor and move assignment operator are affected. I moved both out of the header because they eventually move the unique ptr to the XmlReader/WriterContext (`std::move(other.m_context)`). At the point where this happened in the header, XmlReader/WriterContext is an incomplete type (i.e. basically only forward-declared). In contrast, the definition of XmlReader/WriterContext is visible at the new location in the source file.

(This previously did not matter as `m_context = other.m_context;` copied a raw pointer).